internal/util/addmergecomment/addmergecomment.go

```diff
@@ -41,10 +41,12 @@ func Process(paths ...string) error {
 			Outputs: []kio.Writer{inout},
 		}.Execute()
 		if err != nil {
-			return err
+			// we should do best effort to add merge comment, core Filter method doesn't
+			// return any error, reading resources might throw an error but we should not
+			// error out as we don't need to add merge comments for non-KRM resources
+			return
```

Comment:
I think the filter in this case AddMergeComment should know when to ignore a resource. Right ?

Also I am curious why filters will be invoked on non-KRM resources. I believe we have a check for that.